### PR TITLE
Add optional centering to SimpleShot

### LINKED ISSUE #108: SimpleShot implementation doesn't do centering?
The current implementation of SimpleShot doesn't seem to perform the centering feature transformation, as described in the [paper](https://arxiv.org/pdf/1911.04623.pdf). Currently, it seems the implementation only performs L2 normalization and Euclidean distance (i.e. cosine distance). Are there plans to add the centering transformation?

### PR DESCRIPTION
**Pull request by alanqrwang**
resolves #108

### CODE REVIEW

**Comment by ebennequin** [comment]
Hi @alanqrwang thank you so much for your contribution!

Since feature centering is not specific to SimpleShot and was found to benefit all methods, I added the option at the `FewShotClassifier` level.

I also added the option to normalize the centered features while I was at it (often gives a boost in performance too).

Also, I used the already existing predict_embeddings methods inside your new utils function to avoid duplicated code.

Finally, added some tests.

Please tell me (or update the PR) if you think something is missing!

We'll be waiting for @AntoineToubhans to review before merging, he's coming back next week.

File: easyfsl/methods/utils.py
Comment (AntoineToubhans) [inline easyfsl/methods/utils.py:57]: Do you really need a separate method to do this?
- it seems like it's used only once  -> no need for factorization
- it's quite simple: a one-liner self-explicit formula

**Review by AntoineToubhans — APPROVED** [review]

File: easyfsl/methods/utils.py
Comment (ebennequin) [reply easyfsl/methods/utils.py:57]: I think it can be a useful utils for users who'd want to use power-transform with other FSL methods

**Pull request merged**
